net/tshttpproxy/tshttpproxy_windows.go

```diff
@@ -146,20 +148,33 @@ func proxyFromWinHTTP(ctx context.Context, urlStr string) (proxy *url.URL, err e
 var userAgent = windows.StringToUTF16Ptr("Tailscale")
 
 const (
+	winHTTP_ACCESS_TYPE_DEFAULT_PROXY   = 0
 	winHTTP_ACCESS_TYPE_AUTOMATIC_PROXY = 4
 	winHTTP_AUTOPROXY_ALLOW_AUTOCONFIG  = 0x00000100
 	winHTTP_AUTOPROXY_AUTO_DETECT       = 1
 	winHTTP_AUTO_DETECT_TYPE_DHCP       = 0x00000001
 	winHTTP_AUTO_DETECT_TYPE_DNS_A      = 0x00000002
 )
 
+// Windows 8.1 is actually Windows 6.3 under the hood. Yay, marketing!
+const win8dot1Ver = "6.3"
+
+// We need to use a different flag for proxy resolution depending on whether
+// or not we're running < Windows 8.1
+var proxyFlag = func() uint32 {
```

Comment:
we try to minimize init-time work. It makes binaries start slower and makes the linker unable to discard otherwise unused code.

I'd do this lazily as needed instead. You can use an atomic.Value to cache it between calls to winHTTPOpen.

Reply:
ack